eox_nelp/signals/utils.py

```diff
@@ -0,0 +1,63 @@
+"""Common function for the signals module.
+
+Functions:
+    _generate_external_certificate_data: Generates dict data from CertificateData.
+    _user_has_passing_grade: Determines if the user has a passing grade
+"""
+from django.contrib.auth import get_user_model
+from django.utils import timezone
+from eox_core.edxapp_wrapper.certificates import get_generated_certificate
+from eox_core.edxapp_wrapper.grades import get_course_grade_factory
+from opaque_keys.edx.keys import CourseKey
+
+CourseGradeFactory = get_course_grade_factory()
+GeneratedCertificate = get_generated_certificate()
+User = get_user_model()
+
+
+def _generate_external_certificate_data(timestamp, certificate_data):
```

Comment:
```suggestion
def _generate_futurex_external_certificate_data(timestamp, certificate_data):
```
??
Because we have a function like dispatch futurex progress?

Reply:
I used `futurex` because  that is the name of the service. In this case I have no idea how the service name's but  probably  it's not `futurex` since the base url is different